src/Uno.UWP/Storage/Streams/DataWriter.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using System.IO;
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataWriter : IDataWriter, IDisposable
+	{
+		// only one will be not null - depends on constructor used
+		private MemoryStream _memStream = null;
+		private IOutputStream _outputStream = null;
+		// z: fileOutputStream = detachStream();
+		// OutputStreamOverStream.cs , bez konstruktora, jako io.stream
+
+		private static int _unicodeOrder = 0;
+
+		public UnicodeEncoding UnicodeEncoding { get; set; }
+
+		public ByteOrder ByteOrder { get; set; }
+
+		public uint UnstoredBufferLength
+		{
+			get => (uint)(_memStream.Length - _memStream.Position);
+		}
+
+		//public DataWriterStoreOperation StoreAsync()
+		//{
+		//	// when constructor was without paramaters:
+		//	// System.Runtime.InteropServices.COMException: 'The operation identifier is not valid.
+		//	throw new InvalidOperationException("Windows.Storage.Streams.DataWriter.StoreAsync called, but no Stream was provided in Constructor");
+		//	// else: should do something ...
+		//}
+
+		public IAsyncOperation<bool> FlushAsync() => FlushAsyncTask().AsAsyncOperation<bool>();
+
+		private async Task<bool> FlushAsyncTask()
+		{
+			if (_memStream != null)
+			{
+				await _memStream.FlushAsync();
+				return true;
+			}
+			return true;
+			//if(_outputStream != null)
+			//{
+			//	return await _outputStream.FlushAsync();
+			//}
+		}
+
+		IOutputStream IDataWriter.DetachStream()
+		{
+			// only when constructor was without IOutputStream
+			return null;
+		}
+
+
+		public DataWriter()
+		{
+			_memStream = new MemoryStream();
+		}
+
+		public void WriteByte(byte value)
+		{
+			_memStream.WriteByte(value);
+		}
+
+		public void WriteBytes(byte[] value)
+		{
+			_memStream.Write(value, 0, value.Length);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'value' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5527039)